install/playbooks/roles/system-post-install/tasks/partitions-security.yml

```diff
@@ -0,0 +1,79 @@
+---
+
+# tmp partition ################################################################
+- name: Check if /tmp is a mount point
+  register: tmp_mountpoint_cmd
+  shell: mountpoint -q /tmp/
+  args:
+    warn: false
+
+- name: Mount /tmp as no exec by default
+  when: tmp_mountpoint_cmd.rc == 0
+  register: fstab_file
+  replace:
+    path: /etc/fstab
+    regexp: '(^/dev/[/a-z-]+\s+/tmp\s+ext4\s+)defaults(\s+[0-9]\s+[0-9])'
+    replace: '\1defaults,noexec,nosuid,nodev\2'
+
+- name: Remount /tmp
+  when: fstab_file.changed
+  shell: >-
+    mount -o remount /tmp
+  args:
+    warn: false
+
+- name: Install apt hook to remount /tmp
+  when: fstab_file.changed
+  copy:
+    src: tmp-mnt.cf
+    dest: /etc/apt/apt.conf.d/00tmp-permissions
+
+# boot partition ###############################################################
+- name: Check if /boot is a mount point
+  register: boot_mountpoint_cmd
+  shell: mountpoint -q /boot/
+  args:
+    warn: false
+
+- name: Mount /boot as read only by default
+  when: boot_mountpoint_cmd.rc == 0
+  register: fstab_file
+  replace:
+    path: /etc/fstab
+    regexp: '(^UUID=[0-9a-f-]+\s+/boot\s+ext[2-4]\s+)defaults(\s+[0-9]\s+[0-9])'
+    replace: '\1defaults,ro\2'
+
+- name: Remount /boot
+  when: fstab_file.changed
+  shell: >-
+    mount -o remount /boot
+  args:
+    warn: false
+
+- name: Install apt hook to remount /boot
+  when: fstab_file.changed
+  copy:
+    src: boot-mnt.cf
+    dest: /etc/apt/apt.conf.d/00boot-permissions
+
+# home partition ###############################################################
+- name: Check if /home is a mount point
+  register: home_mountpoint_cmd
+  shell: mountpoint -q /home/
+  args:
+    warn: false
+
+- name: Mount /home as no exec by default
+  when: home_mountpoint_cmd.rc == 0
+  register: fstab_file
+  replace:
+    path: /etc/fstab
+    regexp: '(^/dev/[/a-z-]+\s+/home\s+ext4\s+)defaults(\s+[0-9]\s+[0-9])'
```

Comment:
Thanks,
I intentionally wrote very strict regexp to avoid a system unbootable if an error occurs, _or_ in case the fstab file has been customised.
That said, your suggestion works, so it is worth to consider it, or at least make the regexps more generic. I will.

Reply:
Another issue I discovered, the regex works _all_ the time, even when the partition already has the options you want to add. As a result, the Ansible script wrongly consider the task as modified, and remounts the partition, which is not necessary.
TBH, Ansible is buggy here, as the final file is the same.
Still, I prefer the first option, albeit we can make them more generic (e.g. add btrfs support)

Reply:
Ha, ok, I had tested it with the following and it was working, using ansible 2.8.3.
```
---

- hosts: localhost
  tasks:
    - name: change fstab
      register: fstab
      replace:
        path: /tmp/fstab
        regexp: '(^[^#]\S+\s+/home\s+\S+\s+)\S+(\s+[0-9]\s+[0-9])'
        replace: '\1defaults,noexec,nosuid,nodev\2'

    - name: remount
      debug:
        msg: "fstab changed is {{ fstab.changed }}"
```
If we can't rely on the behavior of Ansible in such a case, then it can come back to what you proposed for the options field, only replacing from the default configuration, like :
```suggestion
    regexp: '(^[^#]\S+\s+/home\s+\S+\s+)defaults(\s+[0-9]\s+[0-9])'
```
which means that if the user has configured mounting options other than `defaults`, it is considered that he knows what he is doing and the playbook won't touch it.

In fact, we want to make sure the options (`noexec`, `nodev`, `nosuid`) are present in the option list, and that they are positioned after `defaults` if it is present, or they replace their positive counterpart if present. I guess that would be a pretty complicated regexp (or several). :sweat_smile:

Reply:
OK, this option is better, I will implement it!